src/collision.cpp

```diff
@@ -37,6 +37,51 @@ bool CheckCollision(const Location_t &Loc1, const Location_t &Loc2)
     return false;
 }
 
+bool CheckCollision(const SpeedlessLocation_t &Loc1, const Location_t &Loc2)
+{
+    if(
+        (Loc1.Y + Loc1.Height >= Loc2.Y) &&
+        (Loc1.Y <= Loc2.Y + Loc2.Height) &&
+        (Loc1.X <= Loc2.X + Loc2.Width) &&
+        (Loc1.X + Loc1.Width >= Loc2.X)
+    )
+    {
+        return true;
+    }
+
+    return false;
+}
```

Comment:
You could simplify this by just returning the result of the condition:
```
return (Loc1.Y + Loc1.Height >= Loc2.Y) &&
       (Loc1.Y <= Loc2.Y + Loc2.Height) &&
       (Loc1.X <= Loc2.X + Loc2.Width) &&
       (Loc1.X + Loc1.Width >= Loc2.X);
```

Reply:
Yes, that is superior. I was just copy-pasting each of the collision calls. This could be a good chance to refactor them.

Reply:
Also done.